docs/audio.rst

```diff
@@ -83,8 +86,216 @@ The built-in sounds can be called using ``audio.play(Sound.NAME)``.
 * ``Sound.TWINKLE``
 * ``Sound.YAWN``
 
+Sounds Example
+--------------
+
+::
+
+    from microbit import *
+
+    while True:
+        if button_a.is_pressed() and button_b.is_pressed():
+            # When pressing both buttons only play via the edge connector
+            audio.play(Sound.HELLO, pin=pin0)
+        elif button_a.is_pressed():
+            # On button A play a sound and when it's done show an image
+            audio.play(Sound.HAPPY)
+            display.show(Image.HAPPY)
+        elif button_b.is_pressed():
+            # On button B play a sound and show an image at the same time
+            audio.play(Sound.TWINKLE, wait=False)
+            display.show(Image.BUTTERFLY)
+
+        sleep(500)
+        display.clear()
+
+
+Sound Effects **V2**
+====================
+
+.. py:class::
+    SoundEffect(preset=None, freq_start=500, freq_end=2500, duration=500, vol_start=255, vol_end=0, wave=WAVE_SQUARE, fx=None, interpolation=INTER_LOG)
+
+    An ``SoundEffect`` instance represents a sound effect, composed by a set of
+    parameters configured via the constructor or attributes.
+
+    All the parameters are optional, with default values as shown above, and
+    they can all be modified via attributes of the same name. For example, we
+    can first create an effect ``my_effect = SoundEffect(duration=1000)``,
+    and then change its attributes ``my_effect.duration = 500``.
+
+    :param preset: An existing SoundEffect instance to use as a base, its values
+        are cloned in the new instance, and any additional arguments provided
+        overwrite the base values.
+    :param freq_start: Start Frequency in Hertz (Hz), eg: ``400``
+    :param freq_end: End Frequency in Hertz (Hz), eg: ``2000``
+    :param duration: Duration of the sound (ms), eg: ``500``
+    :param vol_start: Start volume value, range 0-255, eg: ``120``
+    :param vol_end: End volume value, range 0-255, eg: ``255``
+    :param wave: Type of wave shape, one of these values: ``WAVE_SINE``,
+        ``WAVE_SAWTOOTH``, ``WAVE_TRIANGLE``, ``WAVE_SQUARE``,
+        ``WAVE_NOISE`` (randomly generated noise).
```

Comment:
Where do these `WAVE_SINE` etc constants live?  I see in examples below they are at the top-level in `audio`, eg `audio.WAVE_SINE`.

The other place would be `SoundEffect.WAVE_SINE`.

Reply:
Since built-in sound effects live in `audio.SoundEffect` (like `audio.SoundEffect.CLICK`), it makes sense that these constants live there as well.

Reply:
Yeah, it makes sense to have them inside `audio.SoundEffect`, my worry was that it become too verbose:

```
audio.play(audio.SoundEffect(interpolation=audio.SoundEffect.INTER_LOG))
```

But to be fair, the difference is not that large, and it also helps with the editor autocompletion to not have that much directly inside the audio module.

```
audio.play(audio.SoundEffect(interpolation=audio.INTER_LOG))
```

Reply:
I have added these constants to `audio.SoundEffect`.

Reply:
Okay, I'll modify the docs to reflect that 👍

Reply:
Done in c0a2909120496aa33117d07573c5cd54d18fad60.